include/clock_control.h

```diff
@@ -23,6 +24,20 @@ extern "C" {
 /* Used to select all subsystem of a clock controller */
 #define CLOCK_CONTROL_SUBSYS_ALL	NULL
 
+typedef void (*clock_control_clbk_t)(struct device *dev, void *user_data);
+
+struct clock_control_user {
```

Comment:
name clock_control_async_data would be better. User is whoever uses the whole API.

Reply:
done